### ISSUE TITLE
private_dns_zone_id in azurerm_kubernetes_cluster doesn't support "none"

### ISSUE THREAD

**Issue by SpartyInDetroit** [terraform-providers/terraform-provider-azurerm#10597]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.14.6
+ provider registry.terraform.io/databrickslabs/databricks v0.3.0
+ provider registry.terraform.io/hashicorp/azuread v1.0.0
+ provider registry.terraform.io/hashicorp/azurerm v2.47.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.0.1
+ provider registry.terraform.io/hashicorp/null v3.0.0
+ provider registry.terraform.io/hashicorp/random v3.0.1
+ provider registry.terraform.io/hashicorp/tls v3.0.0

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
resource "azurerm_kubernetes_cluster" "aks" {
  name                    = "${var.environment.code}-Aks-${var.region.short_name}"
  location                = var.region.azure_name
  resource_group_name     = var.resource_group_name
  dns_prefix              = "${var.environment.code}-Aks"
  node_resource_group     = "${var.environment.code}-AksNodes-${var.region.short_name}-RG"
  kubernetes_version      = var.k8s_configuration.version
  private_cluster_enabled = true
  private_dns_zone_id     = "None"
  sku_tier                = "Free"

  service_principal {
    client_id     = var.service_principal_application_id
    client_secret = var.service_principal_secret
  }

  default_node_pool {
    name                  = "general"
    vm_size               = var.k8s_configuration.vm_size
    orchestrator_version  = var.k8s_configuration.version
    max_pods              = "30"
    os_disk_size_gb       = "100"
    enable_node_public_ip = false
    enable_auto_scaling   = true
    min_count             = var.k8s_configuration.min_nodes
    max_count             = var.k8s_configuration.max_nodes
    vnet_subnet_id        = var.subnet_id
    tags                  = merge(var.tags, { "AutoShutdownSchedule" = "AlwaysOn" })
  }

  linux_profile {
    admin_username = "AksAdmin"
    ssh_key {
      key_data = var.ssh_key
    }
  }

  network_profile {
    network_plugin     = "azure"
    docker_bridge_cidr = "172.17.0.1/16"
    network_policy     = "calico"
    service_cidr       = "10.0.0.0/16"
    dns_service_ip     = "10.0.0.10"
  }

  addon_profile {
    kube_dashboard {
      enabled = false
    }

    oms_agent {
      enabled                    = true
      log_analytics_workspace_id = var.log_analytics_workspace_id
    }

    azure_policy {
      enabled = true
    }
  }

  role_based_access_control {
    enabled = true
  }

  tags = merge(var.tags, { "AutoShutdownSchedule" = "AlwaysOn" })
  lifecycle {
    ignore_changes = [
      tags["CreatedDate"],
      default_node_pool[0].tags["CreatedDate"]
    ]
  }
}

```

### Debug Output

https://gist.github.com/SpartyInDetroit/046a884f2e59da4fae4037267708249a

### Expected Behaviour

A new private azure kubernetes service cluster should be created with no private dns zone.

### Actual Behaviour

Terraform throws 2 errors because the azurerm provider is expecting `private_dns_zone_id` to be either `"System"` or an azure id for a BYO private link dns zone.

### Steps to Reproduce

1. Configure `azurerm_kubernetes_cluster`with `private_cluster_enabled = true` and `private_dns_zone_id = "None"`.
2. `terraform plan`

### Important Factoids

This is running in East US 2.

### References

The Azure documentation for this feature can be found here:
https://docs.microsoft.com/en-us/azure/aks/private-clusters#configure-private-dns-zone

**Comment by tombuildsstuff** [comment]
hi @SpartyInDetroit

Terraform uses the convention of omitting fields when a value should be turned off, as such this can be set to `None` by omitting the `private_dns_zone_id` field (although for what it's worth here the Azure/AKS API does the same thing in this instance - not returning this field from the API if it's set to `None`). However at this time it appears that the AKS API requires that a Private DNS Zone is created when creating a Private Cluster, as if this is unspecified (or set to `None`) the API uses the value of `System` - as such it appears that this `System` value is automatically set by Azure.

Since this is a question regarding Terraform Configuration rather than a bug or feature request with the Azure Provider, I'm going to close this issue for the moment - but should you have further questions [I'd direct you towards the community discuss forum](https://discuss.hashicorp.com).

Thanks!

**Comment by SpartyInDetroit** [comment]
@tombuildsstuff

> However at this time it appears that the AKS API requires that a Private DNS Zone is created when creating a Private Cluster, as if this is unspecified (or set to None) the API uses the value of System - as such it appears that this System value is automatically set by Azure.

This is incorrect. In the documentation linked above
https://docs.microsoft.com/en-us/azure/aks/private-clusters#configure-private-dns-zone
Microsoft explicitly states a value of `None` is valid and separate from the default value of `System` which happens if no (or a null) value is provided. This can be confirmed with the az cli with the following command:
`az aks create -n <name> -g <resource-group-name> --load-balancer-sku standard --enable-private-cluster --assign-identity --private-dns-zone None --vnet-subnet-id <subnet-id>`

In summary, `None` and `null` are *not* the same.
